qutip/nonmarkov/bofin.py

```diff
@@ -530,38 +484,77 @@ def boson_grad_prev(self, he_n, k, prev_he):
             )
         return self._pad_op(op, he_n, prev_he)
 
-    def boson_grad_next(self, he_n, k, next_he):
-        """
-        Get the next gradient
-        """
+    def _grad_prev_fermionic(self, he_n, k, prev_he):
+        ck = self.bath.ck
+
+        n_excite = sum(he_n)
```

Comment:
i guess if we want to try allowing bosonic+fermionic baths at the same time, we need a check at here and elsewhere to make sure we only sum over fermionic indices.  maybe this is all? the use of generic dims in the bathmodes stuff helps us do the cut-offs correctly?

Reply:
Yep! The idea is to add `self.bath.is_fermionic` (which is an array for 0 or 1) and then do `sum(he_n[k] * fermionic[k] for k in range(len(he_n))` or something like that.

Probably a good idea to get a test case first before fiddling too much though.